provider/pkg/gen/schema.go

```diff
@@ -970,6 +970,10 @@ func (g *packageGenerator) genResourceVariant(apiSpec *openapi.ResourceSpec, res
 	return nil
 }
 
+func isSingleton(resource *resourceVariant) bool {
+	return resource.PathItem.Delete == nil || customresources.IsSingleton(resource.Path)
+}
```

Comment:
This might be a nice place to explain what a singleton is

```suggestion
// isSingleton indicates if the resource is created and therefore deleted as
// part of its parent resource. When deleting a singleton we will reset its
// properties back to a default state. However a custom resource might still
// provide a delete method to perform a custom reset.
func isSingleton(resource *resourceVariant) bool {
	return resource.PathItem.Delete == nil || customresources.IsSingleton(resource.Path)
}
```